### PR TITLE
chore(amplify_datastore): conditionally add API plugin if api configuration exists

### PR DESCRIPTION
**Pull request by Amplifiyer**
*Description of changes:*
This is intended as a workaround until we have a GraphQL API category in flutter which would be added by the flutter users. This whole thing will go away before GA.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by jamesonwilliams — COMMENTED** [review]
When there is a Flutter GraphQL category, you'll just rely on that being added from Dart, right?

The current workaround is only since there isn't a GraphQL category in Dart, right?

File: packages/amplify_core/ios/Classes/SwiftCore.swift [hunk 1]
```diff
@@ -33,27 +33,26 @@ public class SwiftCore: NSObject, FlutterPlugin {
...
-        } catch {
-            print("Failed to configure Amplify \(error)")
-            result(FlutterError(code: "AmplifyException",
-                                message: "Failed to Configure Amplify",
-                                details: error.localizedDescription));
-        }
+        let arguments = call.arguments as! Dictionary<String, AnyObject>
+        let version = arguments["version"] as! String
+        let configuration = arguments["configuration"] as! String
+        onConfigure(result: result, version: version, configuration: configuration)
```
Comment (haverchuck) [inline packages/amplify_core/ios/Classes/SwiftCore.swift:39]: +1

**Comment by Amplifiyer** [comment]
> When there is a Flutter GraphQL category, you'll just rely on that being added from Dart, right?
> 
> The current workaround is only since there isn't a GraphQL category in Dart, right?

That is correct, I've added some clarification in the description.

**Review by haverchuck — APPROVED** [review]

File: packages/amplify_core/ios/Classes/SwiftCore.swift [hunk 2]
```diff
@@ -33,27 +33,26 @@ public class SwiftCore: NSObject, FlutterPlugin {
...
+      if(!isConfigured) {
         do {
+          if let data = configuration.data(using: .utf8) {
+            let configurationDictionary = try JSONSerialization.jsonObject(with: data) as? [String: Any]
+            if(configurationDictionary?["api"] != nil) {
+                // api configuration exists. Let's add the plugin
+                try Amplify.add(plugin: AWSAPIPlugin())
+            }
+          }
+          let amplifyConfiguration = try JSONDecoder().decode(AmplifyConfiguration.self, from: configuration.data(using: .utf8)!)
```
Comment (Ashish-Nanda) [inline packages/amplify_core/ios/Classes/SwiftCore.swift:55]: You can reuse the variable `data` above.

File: packages/amplify_datastore/ios/amplify_datastore.podspec [hunk 3]
```diff
@@ -17,7 +17,6 @@ The DataStore module for Amplify Flutter.
   s.dependency 'Flutter'
   s.dependency 'Amplify'
   s.dependency 'AmplifyPlugins/AWSDataStorePlugin'
-  s.dependency 'AmplifyPlugins/AWSAPIPlugin'
   s.platform = :ios, '13.0'
```
Comment (Ashish-Nanda) [inline packages/amplify_datastore/ios/amplify_datastore.podspec:20]: Not related to this change, but our platform versions are different in `core` and `datastore`. We should align them.

**Review by Ashish-Nanda — APPROVED** [review]
Minor comments, but looks good.

I would suggest adding tests for this case, but I think this will be removed before GA when we have the API plugin?

File: packages/amplify_datastore/ios/amplify_datastore.podspec (hunk 3 above)
Comment (Amplifiyer) [reply packages/amplify_datastore/ios/amplify_datastore.podspec:20]: No, this is by design. Customers not using DataStore have no reason to have a min iOS 13 version.

**Review by fjnoyp — APPROVED** [review]
Looks good

File: packages/amplify_datastore/ios/amplify_datastore.podspec (hunk 3 above)
Comment (Ashish-Nanda) [reply packages/amplify_datastore/ios/amplify_datastore.podspec:20]: Ah ok, I didn't realize it was intentional.

**Comment by Amplifiyer** [comment]
> Minor comments, but looks good.
> 
> I would suggest adding tests for this case, but I think this will be removed before GA when we have the API plugin?

Yes, as mentioned in the description of this PR.

File: packages/amplify_core/ios/Classes/SwiftCore.swift [hunk 4]
```diff
@@ -33,27 +33,26 @@ public class SwiftCore: NSObject, FlutterPlugin {
...
-          let amplifyConfiguration = try JSONDecoder().decode(AmplifyConfiguration.self, from: configuration.data(using: .utf8)!)
-          onConfigure(result: result, version: version, amplifyConfiguration: amplifyConfiguration)
-            
-        } catch {
-            print("Failed to configure Amplify \(error)")
-            result(FlutterError(code: "AmplifyException",
-                                message: "Failed to Configure Amplify",
-                                details: error.localizedDescription));
-        }
+        let arguments = call.arguments as! Dictionary<String, AnyObject>
```
Comment (royjit) [inline packages/amplify_core/ios/Classes/SwiftCore.swift:36]: nit: These values never fail? ie call.arguments always contains Dictionary? I am concerned with the force casting. 
If this is intended, I guess adding some comments would make it clear.

**Review by royjit — APPROVED** [review]
LGTM: Added minor comment

File: packages/amplify_core/ios/Classes/SwiftCore.swift (hunk 4 above)
Comment (Amplifiyer) [reply packages/amplify_core/ios/Classes/SwiftCore.swift:36]: yeah we need to put guards here. This is an earlier category that we didn't add much error handling. We will be addressing this in future PRs.

**Pull request merged**